internal/venice-common/src/main/java/com/linkedin/venice/views/VeniceView.java

```diff
@@ -106,7 +107,50 @@ public static int parseVersionFromViewTopic(String topicName) {
       throw new VeniceException("Cannot parse version because this is not a view topic, topic name: " + topicName);
     }
     int versionStartIndex = Version.getLastIndexOfVersionSeparator(topicName) + Version.VERSION_SEPARATOR.length();
-    int versionEndIndex = versionStartIndex + topicName.substring(versionStartIndex).indexOf(VIEW_TOPIC_SEPARATOR);
+    int versionEndIndex = versionStartIndex + topicName.substring(versionStartIndex).indexOf(VIEW_NAME_SEPARATOR);
     return Integer.parseInt(topicName.substring(versionStartIndex, versionEndIndex));
   }
+
+  /**
+   * @param topicName for the view topic
+   * @return the corresponding store name and view name for the given view topic in a single string.
+   */
+  public static String parseStoreAndViewFromViewTopic(String topicName) {
+    String storeName = parseStoreFromViewTopic(topicName);
+    int viewTopicSuffixIndex = topicName.lastIndexOf(VIEW_NAME_SEPARATOR);
+    int viewNameStartIndex = topicName.substring(0, viewTopicSuffixIndex).lastIndexOf(VIEW_NAME_SEPARATOR);
+    return VIEW_STORE_PREFIX + storeName + topicName.substring(viewNameStartIndex, viewTopicSuffixIndex);
+  }
+
+  public static String getViewStoreName(String storeName, String viewName) {
+    return VIEW_STORE_PREFIX + storeName + VIEW_NAME_SEPARATOR + viewName;
+  }
+
+  public static String getStoreNameFromViewStoreName(String viewStoreName) {
+    String viewStoreNameWithoutPrefix = viewStoreName.substring(VIEW_STORE_PREFIX.length());
+    int storeNameEndIndex = viewStoreNameWithoutPrefix.lastIndexOf(VIEW_NAME_SEPARATOR);
+    return viewStoreNameWithoutPrefix.substring(0, storeNameEndIndex);
+  }
+
+  public static String getViewNameFromViewStoreName(String viewStoreName) {
+    int viewNameStartIndex = viewStoreName.lastIndexOf(VIEW_NAME_SEPARATOR);
+    return viewStoreName.substring(viewNameStartIndex + 1);
+  }
+
+  public static boolean isViewStore(String storeName) {
+    return storeName.startsWith(VIEW_STORE_PREFIX);
```

Comment:
Can these util methods handle `non-view` params?

Reply:
Good catch, and the answer is no. I will add `isViewStore` checks and throw exceptions to the ones that is expecting a view store name.